subtests/docker_cli/run_cgroups/run_cgroups.py

```diff
@@ -55,3 +56,41 @@ class memory_no_cgroup(memory_base):
 
 class memory_negative(memory_base):
     pass
+
+
+class cgroup_parent_invalid_name(cgroup_parent_base):
+    """
+    Invalid value for --cgroup-parent option
+    """
+
+    def initialize(self):
+        super(cgroup_parent_invalid_name, self).initialize()
+        self._setup("/foobar",
+                    "",
+                    "Error response from daemon: Cannot start container {cid}:"
+                    " [8] System error: Invalid slice name /foobar")
+
+
+class cgroup_parent_path(cgroup_parent_base):
+    """
+    Perfectly valid slice. In docker < 1.9.1-25.el7 this produced
+    weird results.
+    """
+
+    def initialize(self):
+        super(cgroup_parent_path, self).initialize()
+        self._setup("{rand1}.slice",
+                    "/{rand1}.slice/docker-{cid}.scope",
+                    "")
+
+
+class cgroup_parent_path_with_hyphens(cgroup_parent_base):
+    """
+    Hyphens result in a more complicated path.
+    """
+
+    def initialize(self):
+        super(cgroup_parent_path_with_hyphens, self).initialize()
+        self._setup("{rand1}-{rand2}.slice",
+                    "/{rand1}.slice/{rand1}-{rand2}.slice/docker-{cid}.scope",
+                    "")
```

Comment:
I think this is fine, I agree it's a little hard to follow and keep straight, but it's a rush-request and plenty of other tests do it this way.  If you want to try a different approach on a less-urgent test, please feel free, I'm happy to take a look.